lib/archive_validator/author_private_key_validator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class ArchiveValidator
+  class AuthorPrivateKeyValidator < BaseValidator
+    def validate
+      return if person.nil?
+      return if person.serialized_public_key == private_key.public_key.export
+
+      messages.push("Private key in the archive doesn't match the known key of #{person.diaspora_handle}")
+    rescue DiasporaFederation::Discovery::DiscoveryError
```

Comment:
Since this is a critical validation, why is it ok to completely ignore this error?

Reply:
`person` does `Person.find_or_fetch_by_identifier` if the person is unknown. But this validation should only block if the person is known with a different key. So for example:

I import account `a@a` to pod `b`, which already knows `a@a`, if the archive then contains the wrong key for `a@a`, this should fail. If I want to import it to pod `c`, which doesn't know `a@a` yet, and pod `a` is down, it should still be possible to import the archive. It then sends a migration-message to other pods signed with the key from the archive and every other pod knowing `a@a` can validate it and if it's valid, process the migration message.

Reply:
But it probably should be logged or a comment should be added, otherwise it could be difficult to still know this in the future :)

(Also I forgot the case where pod `a` is up, and pod `c` doesn't know the user, then the user is fetched, not `DiscoveryError` happens, and if the key is wrong, this fails as it should :) )

Reply:
Yeah, we don't actually ignore this error. If migration failed to discover the archive author the rake command output will be different and the user will notice a warning. It's just not a responsibility of the validator to explain this issue to the user.